src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java

```diff
@@ -261,12 +269,131 @@ private static File createInOrderSampleMap() {
+        multipleOutOfOrderRenamingsAcrossBatches.put("someOtherSample", "s4");
+
+
+        final List<Integer> batchSizes = Arrays.asList(0, 1, 4);
+        final List<Integer> threads = Arrays.asList(1, 2);
+        final List<Map<String, String>> mappings = Arrays.asList(noRemapping, sameInput, sameInputWeirdOrder, swizzled, multipleOutOfOrderRenamingsAcrossBatches);
+        final List<Object[]> out = new ArrayList<>();
+        for(final Map<String,String> mapping : mappings){
+            for(final int batchSize :batchSizes){
+                for(final int threading : threads){
+                    out.add( new Object[]{mapping, threading, batchSize});
+                }
+            }
+        }
+        return out.iterator();
+    }
+
+
+    @Test(dataProvider = "getRenameCombinations")
+    public void testRenamingSamples(final Map<String, String> renamingMap, final int threads, final int batchSize) throws IOException {
+        final LinkedHashMap<String, String> sampleMap = new LinkedHashMap<>(renamingMap);
+        sampleMap.replaceAll( (newSampleName, originalSampleName)-> createInputVCF(originalSampleName).getAbsolutePath());
+
+        final File sampleMapFile = getSampleMapFile(sampleMap);
+
+        final String workspace = createTempDir("workspace").getAbsolutePath();
+        Files.delete(Paths.get(workspace));
+        final ArgumentsBuilder args = new ArgumentsBuilder()
+                .addArgument(GenomicsDBImport.SAMPLE_NAME_MAP_LONG_NAME, sampleMapFile.getAbsolutePath())
+                .addArgument(GenomicsDBImport.WORKSPACE_ARG_NAME, new File(workspace).getAbsolutePath())
+                .addArgument(GenomicsDBImport.VCF_INITIALIZER_THREADS_LONG_NAME, String.valueOf(threads))
+                .addArgument(GenomicsDBImport.BATCHSIZE_ARG_NAME, String.valueOf(batchSize))
+                .addArgument("L", IntervalUtils.locatableToString(INTERVAL));
+
+        runCommandLine(args);
+        final Set<String> expectedSampleNames = sampleMap.keySet();
+        try(final GenomicsDBFeatureReader<VariantContext, PositionalBufferedStream> reader = getGenomicsDBFeatureReader(workspace, b37_reference_20_21)) {
+            final CloseableTribbleIterator<VariantContext> iterator = reader.iterator();
+            Assert.assertTrue(iterator.hasNext(), "expected to see a variant");
+            Assert.assertTrue(expectedSampleNames.size() > 0);
+            Assert.assertEquals(expectedSampleNames.size(), renamingMap.size());
+            iterator.forEachRemaining(vc -> {
+                   Assert.assertEquals(vc.getSampleNames(), expectedSampleNames);
+                expectedSampleNames.forEach( sample -> {
+                   Assert.assertEquals(vc.getGenotype(sample).getAnyAttribute(SAMPLE_NAME_KEY), renamingMap.get(sample));
+                   Assert.assertEquals(vc.getGenotype(sample).getAnyAttribute(ANOTHER_ATTRIBUTE_KEY), 10); //check another attribute just to make sure we're not mangling things
```

Comment:
Have you confirmed that this test case fails without the fix to the tool?

Reply:
A subset of the test cases fail without the fix.  It only applies to 1 code path and it depends on the batching, so many pass with / without the fix.